java/cdk/src/main/java/sleeper/cdk/stack/CompactionStack.java

```diff
@@ -157,7 +158,8 @@ public CompactionStack(
             InstanceProperties instanceProperties,
             BuiltJars jars,
             Topic topic,
-            CoreStacks coreStacks) {
+            CoreStacks coreStacks,
+            Consumer<IMetric> errorMetricsConsumer) {
```

Comment:
It might be a bit easier to read if we just pass the list around the stacks rather than a consumer?